lua/gitlinker.lua

```diff
@@ -100,7 +100,7 @@ local function get_buf_range_url_data(mode, user_opts)
   end
 
   return vim.tbl_extend("force", repo_url_data, {
-    rev = rev,
+    rev = repo_url_data.rev or rev,
```

Comment:
we wouldn't have to do this if the key was called `branch`. Then we can embed both the branch name and the rev sha in the url data that is passed to callbacks.

Which is good because then it is up to the callbacks to decide what they want to do (either use the branch name or the rev sha).